socialhome/users/views.py

```diff
@@ -67,18 +68,11 @@ def get_context_data(self, **kwargs):
 
     def get_json_context(self):
         json_context = super().get_json_context()
-        json_context.update({
-            "profile": {
-                "id": self.target_profile.id,
-                "guid": self.target_profile.guid,
-                "followersCount": self.followers_count,
-                "followingCount": str(self.target_profile.following.count()),
-                "handle": self.target_profile.handle,
-                "saferImageUrlLarge": self.target_profile.safer_image_url_large,
-                "streamType": self.profile_stream_type,
-                "pinnedContentExists": self.pinned_content_exists,
-            },
-        })
+        if self.request.user.is_authenticated:
+            json_context.update({"profile": ProfileSerializer(self.target_profile).data})
```

Comment:
I guess you're right about using the profile serializer to serialize the data for the frontend, but the trouble with adding all the extra fields there is that it will make the profile API a lot slower. But, since we disabled profile list API view totally some time ago, I guess performance isn't really an issue.

However, using `LimitedProfileSerializer` for unauthenticated users is wrong. It's not meant to control visibility, but to contain less data, ie the data required to render a grid item. A serializer doesn't itself do anything about visibility - that is what the view is for. If using a serializer directly, one must ensure only those fields are coming out that the user can see.

Out `ProfileSerializer` contains only data that is available always if the user can see the profile, so this is ok. Just remove the `else` and the usage of the limited serializer.

Reply:
> But, since we disabled profile list API view totally some time ago, I guess performance isn't really an issue.

I prefer using one unique source rather than multiple. If there's a serializer for the API, better using it. What the front-end needs, any other application may need it. So, is it a go or a no-go?

Reply:
Sure, but we must also think of performance. As said, this is fine, since we only fetch individual objects, not lists. If we do in the future, we'll need to precalculate the values on the model. For now, this is fine.